fiftyone/brain/internal/core/leaky_splits.py

```diff
@@ -0,0 +1,435 @@
+                    **self.config.similarity_config_dict,
+                    **sim_conf_dict_aux,
+                }
+
+            similarity_config_object = sim._parse_config(**sim_conf_dict_aux)
+            self._similarity_method = similarity_config_object.build()
+            self._similarity_method.ensure_requirements()
+
+            if self._save_similarity_index:
+                self._similarity_method.register_run(
+                    samples,
+                    self.config.similarity_brain_key,
+                )
+
+            self._similarity_index = self._similarity_method.initialize(
+                samples, self.config.similarity_brain_key
+            )
+
+    def set_threshold(self, threshold):
+        """Set threshold for leak computation"""
+        self._leak_threshold = threshold
+
+    @property
+    def leaks(self):
+        """
+        Returns view with all potential leaks.
+        """
+
+        # access cache if possible
+        if self._last_computed_threshold == self._leak_threshold:
+            return self._leaks
+
+        # populate index if it doesn't have some samples
+        if not self._similarity_index.total_index_size == len(self.samples):
+            (
+                embeddings,
+                sample_ids,
+                label_ids,
+            ) = self._similarity_index.compute_embeddings(self.samples)
+            self._similarity_index.add_to_index(
+                embeddings, sample_ids, label_ids
+            )
+
+        # check if duplicates already computed in the index
+        if not self._similarity_index.thresh == self._leak_threshold:
+            self._similarity_index.find_duplicates(self._leak_threshold)
+        duplicates = self._similarity_index.duplicates_view()
+
+        # filter duplicates to just those with neighbors in different splits
+        to_keep = []
+        for (
+            sample_id,
+            neighbors,
+        ) in self._similarity_index.neighbors_map.items():
+            keep_sample = False
+            sample_split = self.id2split[sample_id]
+            leaks = []
+            for n in neighbors:
+                if not (self.id2split[n[0]] == sample_split):
+                    # at least one of the neighbors is from a different split
+                    # we keep this one
+                    keep_sample = True
+                    leaks.append(n[0])
+
+            if keep_sample:
+                to_keep.append(sample_id)
+                # remove all other samples because they are all from the same split
+                to_keep = to_keep + leaks
+
+        duplicates = duplicates.select(to_keep)
+
+        # cache to avoid recomputation
+        self._last_computed_threshold = self._leak_threshold
+        self._leaks = duplicates
+
+        return duplicates
+
+    @property
+    def num_leaks(self):
+        """Returns the number of leaks found."""
+        return self.leaks.count
+
+    def leaks_by_sample(self, sample):
```

Comment:
I would expect something called `leaks_by_samples` to return a dictionary, not give the leaks for a single sample.

Reply:
That may be useful functionality and something I can add. It's effectively what the main `leaks` property gives but in view form. What would you recommend changing this method's name to?

Reply:
Maybe `leaks_for_sample()`?